src/logic/wallets/getWeb3.ts

```diff
@@ -146,6 +147,8 @@ export const reverseENSLookup = async (address: string): Promise<string> => {
   return isValidAddress ? name : ''
 }
 
+export const getDomainPart = (name: string): string => name.split('.')[0]
```

Comment:
It's better to remove the last part, not pick the first one. ENS domains can have subdomains.
```
name.replace(/\.[^.]+$/, '')
```